lib/charms/loki_k8s/v1/loki_push_api.py

```diff
@@ -2748,3 +2752,55 @@ def _exec(self, cmd) -> str:
         result = subprocess.run(cmd, check=True, stdout=subprocess.PIPE)
         output = result.stdout.decode("utf-8").strip()
         return output
+
+
+def charm_logging_config(
+    endpoint_requirer: LokiPushApiConsumer, cert_path: Optional[Union[Path, str]]
+) -> Tuple[Optional[List[str]], Optional[str]]:
+    """Utility function to determine the charm_logging config you will likely want.
+
+    If no endpoint is provided:
+     disable charm logging.
+    If https endpoint is provided but cert_path is not found on disk:
+     disable charm logging.
+    If https endpoint is provided and cert_path is None:
+     ERROR
+    Else:
+     proceed with charm logging (with or without tls, as appropriate)
+    Usage:
+    >>> from lib.charms.loki_k8s.v0.charm_logging import log_charm
+    >>> from lib.charms.loki_k8s.v1.loki_push_api import charm_logging_config
+    >>> @log_charm(logging_endpoint="my_endpoints", cert_path="cert_path")
+    >>> class MyCharm(...):
+    >>>     _cert_path = "/path/to/cert/on/charm/container.crt"
+    >>>     def __init__(self, ...):
+    >>>         self.logging = LokiPushApiConsumer(...)
+    >>>         self._my_endpoints, self._cert_path = charm_logging_config(
+    ...             self.logging, self._cert_path)
+    >>>     @property
+    >>>     def my_endpoints(self):
+    >>>         return self._my_endpoints
+    >>>     @property
+    >>>     def cert_path(self):
+    >>>         return self._cert_path
+    """
```

Comment:
As we are doing in the libs, I would place the concrete example and some explanation in the initial docstring of the file.

About the example itself: In `log_charm` function the first argument is `logging_endpoints` and the second one is `server_cert`

Reply:
ach, silly me